src/erl_optics_lens.erl

```diff
@@ -20,7 +20,7 @@
     update/2
 ]).
 
--type histo_buckets() :: list(float()).
+-type histo_buckets() :: list(tuple()).
```

Comment:
This should not typecheck. The spec for `histo/2` says it accepts `[integer()]`, not `[tuple()]`.

Also, you can specify complex types in type specs. If I understand your code correctly, the type should be `[{integer(), integer()}]`.